llvm/test/Transforms/InstCombine/cheri-memcpy-64bit-cap.ll

```diff
@@ -0,0 +1,84 @@
+; CHECK-LABEL: @test_memcpy8_align_4(
+; CHECK-NEXT:  entry:
+; CHECK-NEXT:    call void @llvm.memcpy.p200.p200.i32(ptr addrspace(200) noundef nonnull align 4 dereferenceable(8) [[D:%.*]], ptr addrspace(200) noundef nonnull align 4 dereferenceable(8) [[S:%.*]], i32 8, i1 false) #[[ATTR1]]
+; CHECK-NEXT:    ret ptr addrspace(200) [[D]]
+;
+entry:
+  ; memcpy cannot be transformed, it may contain a capability, but alignment is insufficient
+  call void @llvm.memcpy.p200.p200.i32(ptr addrspace(200) align 4 %d, ptr addrspace(200) align 4 %s, i32 8, i1 false) must_preserve_cheri_tags
+  ret ptr addrspace(200) %d
+}
+
+define ptr addrspace(200) @test_memcpy8_nocap(ptr addrspace(200) noundef %d, ptr addrspace(200) noundef %s) local_unnamed_addr addrspace(200) {
+; CHECK-LABEL: @test_memcpy8_nocap(
+; CHECK-NEXT:  entry:
+; CHECK-NEXT:    [[TMP0:%.*]] = load i64, ptr addrspace(200) [[S:%.*]], align 8
+; CHECK-NEXT:    store i64 [[TMP0]], ptr addrspace(200) [[D:%.*]], align 8
+; CHECK-NEXT:    ret ptr addrspace(200) [[D]]
+;
+entry:
+  ; memcpy does not copy capabilities, so can be transformed to regular load/store
+  call void @llvm.memcpy.p200.p200.i32(ptr addrspace(200) align 8 %d, ptr addrspace(200) align 8 %s, i32 8, i1 false) no_preserve_cheri_tags
+  ret ptr addrspace(200) %d
+}
+
+define ptr addrspace(200) @test_memcpy8_nocap_align_unkown(ptr addrspace(200) noundef %d, ptr addrspace(200) noundef %s) local_unnamed_addr addrspace(200) {
+; CHECK-LABEL: @test_memcpy8_nocap_align_unkown(
+; CHECK-NEXT:  entry:
+; CHECK-NEXT:    [[TMP0:%.*]] = load i64, ptr addrspace(200) [[S:%.*]], align 1
+; CHECK-NEXT:    store i64 [[TMP0]], ptr addrspace(200) [[D:%.*]], align 1
+; CHECK-NEXT:    ret ptr addrspace(200) [[D]]
+;
+entry:
+  ; memcpy has unknown alignment but does not copy capabilities, so can be transformed to regular load/store
+  call void @llvm.memcpy.p200.p200.i32(ptr addrspace(200) %d, ptr addrspace(200) %s, i32 8, i1 false) no_preserve_cheri_tags
+  ret ptr addrspace(200) %d
+}
+
+define ptr addrspace(200) @test_memcpy8_nocap_align_4(ptr addrspace(200) noundef %d, ptr addrspace(200) noundef %s) local_unnamed_addr addrspace(200) {
+; CHECK-LABEL: @test_memcpy8_nocap_align_4(
+; CHECK-NEXT:  entry:
+; CHECK-NEXT:    [[TMP0:%.*]] = load i64, ptr addrspace(200) [[S:%.*]], align 4
+; CHECK-NEXT:    store i64 [[TMP0]], ptr addrspace(200) [[D:%.*]], align 4
+; CHECK-NEXT:    ret ptr addrspace(200) [[D]]
+;
+entry:
+  ; memcpy does not copy capabilities, so can be transformed to regular load/store
+  call void @llvm.memcpy.p200.p200.i32(ptr addrspace(200) align 4 %d, ptr addrspace(200) align 4 %s, i32 8, i1 false) no_preserve_cheri_tags
+  ret ptr addrspace(200) %d
+}
+
+; Function Attrs: argmemonly mustprogress nocallback nofree nounwind willreturn
+declare void @llvm.memcpy.p200.p200.i32(ptr addrspace(200) noalias nocapture writeonly, ptr addrspace(200) noalias nocapture readonly, i32, i1 immarg) addrspace(200) #1
```

Comment:
Even all these attributes should be removable, LLVM will add them automatically

Reply:
Should be fixed now?